### ISSUE TITLE
[HOLD for payment 2023-09-20] [HOLD for payment 2023-09-18][$1000] Inconsistency: Different grayness levels on parent and child message

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#24263]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Be Offline
2. Go to any chat
3. Send a message
4. Go reply in thread
5. send a reply
Notice the level of grayness on the Parent and child message
## Expected Result: 
Same grayness levels
## Actual Result: 
Different grayness levels

## Workaround:
Can the user still use Expensify without this being fixed? Have you informed them of the workaround?

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [x] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** 1.3.51-0
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation

https://github.com/Expensify/App/assets/43996225/cefddbc7-4b45-4d61-9902-361514c8935b

![Snip - (9) New Expensify - Google Chrome](https://github.com/Expensify/App/assets/43996225/f26ae0ac-9f07-445a-b60e-5b9190077554)

**Expensify/Expensify Issue URL:**
**Issue reported by:** @daveSeife
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1690943139195889

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~0192daee63c42d4dbd</li>
        <li>Upwork Job ID: 1689013073676578816</li>
        <li>Last Price Increase: 2023-08-15</li>
<li>Automatic offers: </li>
<ul>
<li>ginsuma | Contributor | 26258577</li>
<li>daveSeife | Reporter | 26258578</li>
</ul></li>
    </ul>
</details>

**Comment by joekaufmanexpensify** [comment]
I can reproduce this. I can't think of any reason why we'd want parent and child messages to be in a different shade of grey here. Especially as the parent and child messages are normally the same text color while online. 


https://github.com/Expensify/App/assets/54866469/2e943554-567e-42bb-8736-d7dd85ba4188

**Comment by saqixpro** [comment]
Hi I can work on this,

I'm already a contributor and here are my details

Expensify Email: saqib@itdunya.com
Upwork Link: www.upwork.com/freelancers/saqixpro 

I have a question, which grey do you think is the one we should stick to? As per the screenshot?

**Comment by ares-dev05** [comment]
## Proposal
### Please re-state the problem that we are trying to solve in this issue.
Different grayness levels
### What is the root cause of that problem?
![image](https://github.com/Expensify/App/assets/83194791/4e4efd2b-b7e6-4cf4-ae19-618a05eec03b)
When you are offline status, the parent characters have opacity 0.5 but the child characters have opacity 1.

This is why `needsOpacity` in OfflineWithFeedback.js is false in the child message component.
### What changes do you think we should make in order to solve the problem?
So we have to enable `needsOpacity` to true in the child OfflineWithFeedback.js as well. For it, we can update the line 99 in OfflineWithFeedback.js as the following.

`const needsOpacity = (isOfflinePendingAction && !isUpdateOrDeleteError) || isAddError || props.network.isOffline;`
### What alternative solutions did you explore? (Optional)

**Comment by HenryHover** [comment]
## Proposal
### Please re-state the problem that we are trying to solve in this issue.
The level of grayness on the Parent and child messages are different while offline

### What is the root cause of that problem?
https://github.com/Expensify/App/blob/main/src/pages/home/report/ReportActionItemParentAction.js#L55-L75

ReportActionItemParentAction component renders ReportActionItem component wrapped by OfflineWithFeedback component

https://github.com/Expensify/App/blob/811a83e28b98a7c348081fc0dbfe1f658c57f1d0/src/pages/home/report/ReportActionItem.js#L533-L564

And ReportActionItem also renders content wrapped by OfflineWithFeedback component

So the parent report content is wrapped by 2 OfflineWithFeedback Components, thereby we have darker grey

### What changes do you think we should make in order to solve the problem?
We should make ReportActionItemParentAction has only one OfflineWithFeedback component
Removing OfflineWithFeedback component from ReportActionItemParentAction can fix the issue visually but it is not good solution I think

As I checked OfflineWithFeedback component should show error messages when the request is not successfully sent
And when user reacts to these error messages, the handling procedure is different for Parent and Child Reports

For Parent Report
https://github.com/Expensify/App/blob/811a83e28b98a7c348081fc0dbfe1f658c57f1d0/src/pages/home/report/ReportActionItemParentAction.js#L59
For Child Report
https://github.com/Expensify/App/blob/811a83e28b98a7c348081fc0dbfe1f658c57f1d0/src/pages/home/report/ReportActionItem.js#L534

I would like to discuss about the project in detail with the main developers

**Comment by joekaufmanexpensify** [comment]
@sobitneupane thoughts on these proposals?

**Comment by joekaufmanexpensify** [comment]
Bumped in Slack.

**Comment by sobitneupane** [comment]
Thanks for the [proposal](https://github.com/Expensify/App/issues/24263#issuecomment-1669655689) @ginsuma

> The `ReportActionItemParentAction` content is wrapped by two `OfflineWithFeedback`

If `ReportActionItemParentAction` is wrapped by two `OfflineWithFeedback`, rather than introducing a new prop can we simply remove `OfflineWithFeedback` from `ReportActionItemParentAction` as suggested by @tienifr in his [proposal](https://github.com/Expensify/App/issues/24263#issuecomment-1669567526).

**Comment by joekaufmanexpensify** [comment]
@sobitneupane do you think we're close to being able to use any of these proposals?

**Comment by DylanDylann** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Inconsistency: Different grayness levels on parent and child message

### What is the root cause of that problem?
We are applying 2 OfflineWithFeedback  to the first message of child thread
https://github.com/Expensify/App/blob/7f13cdd9f5ade3aed50827084aa49b5e3a6496e1/src/pages/home/report/ReportActionItemParentAction.js#L55-L60 
https://github.com/Expensify/App/blob/811a83e28b98a7c348081fc0dbfe1f658c57f1d0/src/pages/home/report/ReportActionItem.js#L533-L539

### What changes do you think we should make in order to solve the problem?
We should remove OfflineWithFeedback in ReportActionItemParentAction. And then, in ReportActionItemParentAction we will pass new prop called offlineObject like this:
```
offlineObject: {
            pendingAction={lodashGet(props.report, 'pendingFields.addWorkspaceRoom') || lodashGet(props.report, 'pendingFields.createChat')}
            errors={lodashGet(props.report, 'errorFields.addWorkspaceRoom') || lodashGet(props.report, 'errorFields.createChat')}
            onClose={() => Report.navigateToConciergeChatAndDeleteReport(props.report.reportID)}
}
```

And in ReportActionItem, if offlineObject props exist we will use it else we still use the current condition

### What alternative solutions did you explore? (Optional)

**Comment by sobitneupane** [comment]
Thanks for the update @ginsuma.

I don't think we will like to show two different errors for the same `reportactionitem`. Were you able to reproduce any such issue?

**Comment by kaushiktd** [comment]
I've checked the solution given by @ginsuma and it's not working for me. Maybe I'm missing something here?

https://drive.google.com/file/d/14tGSAM7aPmQomalluI2DiXMMRHcv-9Oz/view?usp=sharing

**Comment by DylanDylann** [comment]
@sobitneupane What do you think about [my proposal](https://github.com/Expensify/App/issues/24263#issuecomment-1680418112)

**Comment by sobitneupane** [comment]
Thanks for the update @ginsuma. Can you please add some explanation about the proposed condition `_.isEmpty(parentReportAction.errors) && parentReportAction.pendingAction` in your proposal.

**Comment by DylanDylann** [comment]
@sobitneupane 
Could you check [my proposal](https://github.com/Expensify/App/issues/24263#issuecomment-1680418112) ?
I think if there are 2 errors at the same time, we only should display one of them

**Comment by kaushiktd** [comment]
> I've checked the solution given by @ginsuma and it's not working for me. Maybe I'm missing something here?
> 
> https://drive.google.com/file/d/14tGSAM7aPmQomalluI2DiXMMRHcv-9Oz/view?usp=sharing

@sobitneupane I hope you looked at this.

**Comment by joekaufmanexpensify** [comment]
Not overdue.

**Comment by sobitneupane** [comment]
@DylanDylann Those two are different errors and I believe it was intentionally added in `ReportActionItemParentAction`. We handle those errors from `ReportActionItemCreated` for other reports like one-one chat, rooms, etc. But thread report doesn't not have a separate header message unlike `ReportActionItemCreated`.

**Comment by sobitneupane** [comment]
Thanks for the update @ginsuma.

[Proposal](https://github.com/Expensify/App/issues/24263#issuecomment-1669655689) from @ginsuma looks good to me.

🎀 👀 🎀 C+ reviewed

**Comment by kaushiktd** [comment]
@sobitneupane Not sure why you are ignoring my message(s)? I mean I don't mind who gets hired but I'm trying to learn here what I missed so that I can improve myself.

**Comment by sobitneupane** [comment]
@kaushiktd I went through the proposal. And it's working fine on my end. You can try again with the suggested changes in the proposal.

**Comment by kaushiktd** [comment]
@ginsuma yes and that is exactly what I'm trying to learn if you can help.

**Comment by pecanoro** [comment]
Just noting that I reverted the PR [here](https://github.com/Expensify/App/pull/26211) to fix the deploy blocker, a new PR that fixes the bug and does not cause the same regression needs to be created again!

**Comment by joekaufmanexpensify** [comment]
@sobitneupane mind completing the BZ checklist here so we can prep to issue payment next week?

**Comment by pecanoro** [comment]
@joekaufmanexpensify There was a regression, so I reverted the PR. The new PR is on HOLD because of the feature freeze, so we probably need to hold payment until we can merge.

**Comment by joekaufmanexpensify** [comment]
Ah, okay makes sense. I'll hold payment for 7 days after [this PR](https://github.com/Expensify/App/pull/26213) is deployed to Prod. Thanks @pecanoro

**Comment by joekaufmanexpensify** [comment]
New PR is still held. Removing payment date from title for now.

**Comment by joekaufmanexpensify** [comment]
Looks like we may merge the PR soon as the merge freeze is over.

**Comment by techievivek** [comment]
@joekaufmanexpensify Merge freeze is over now https://expensify.slack.com/archives/C01GTK53T8Q/p1694405959277169 so I went ahead and merged the PR.

**Comment by joekaufmanexpensify** [comment]
@sobitneupane mind completing the BZ checklist here so we can prep to issue payment?

**Comment by joekaufmanexpensify** [comment]
Bump @sobitneupane mind working on the BZ checklist so we can finish this one off?

**Comment by joekaufmanexpensify** [comment]
Bumped in Slack

**Comment by sobitneupane** [comment]
> **BugZero Checklist:** The PR fixing this issue has been merged! The following checklist ([instructions](https://stackoverflow.com/c/expensify/questions/14574)) will need to be completed before the issue can be closed:
> 
> * [ ]  [@sobitneupane] The PR that introduced the bug has been identified. Link to the PR:

https://github.com/Expensify/App/pull/18522
> * [ ]  [@sobitneupane] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment:

https://github.com/Expensify/App/pull/18522#issuecomment-1732860004
> * [ ]  [@sobitneupane] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion:

Probably offline tests were not performed in the linked PR. This should have been caught in offline test.

> * [ ]  [@sobitneupane] Determine if we should create a regression test for this bug.

Yes.
> * [ ]  [@sobitneupane] If we decide to create a regression test for the bug, please propose the [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps to ensure the same bug will not reach production again.

**Comment by sobitneupane** [comment]
**Regression Test Proposal**

- Be Offline
- Go to any chat
- Send a message
- Reply in thread
- Send a reply and verify that there is no difference in the grayness between parent and child messages.

**Comment by sobitneupane** [comment]
I will request payment in newDot once the payment is summarized.

cc: @joekaufmanexpensify

**Comment by joekaufmanexpensify** [comment]
Great, thanks! I will work on the BZ checklist.

**Comment by joekaufmanexpensify** [comment]
IMO, this is a very minor visual bug, and no regression test is needed.

**Comment by joekaufmanexpensify** [comment]
BZ checklist is complete, all set to issue payment. We need to make the following payments: 

- @daveSeife - $250 for report. Paid via Upwork. 
- @ginsuma - $500 for PR. (Originally $1,000 and reduced 50% because of regression. Regression also negates urgency bonus.) Paid via Upwork. 
- @sobitneupane - $500 for C+. (Originally $1,000 and reduced 50% because of regression. Regression also negates urgency bonus.) Paid via NewDot.

**Comment by joekaufmanexpensify** [comment]
@daveSeife $250 sent and contract ended!

**Comment by joekaufmanexpensify** [comment]
@ginsuma $500 sent and contract ended!

**Comment by joekaufmanexpensify** [comment]
@sobitneupane could you please request $500 and confirm here once that's done? TY!

**Comment by daveSeife** [comment]
@joekaufmanexpensify received. Thank you!

**Comment by joekaufmanexpensify** [comment]
Pending request

**Comment by joekaufmanexpensify** [comment]
Closing per our slack discussion! Payment summary message for when @sobitneupane requests is [here. ](https://github.com/Expensify/App/issues/24263#issuecomment-1736628010)

**Comment by sobitneupane** [comment]
https://github.com/Expensify/App/issues/24263#issuecomment-1736628010

Requested payment on newDot

**Comment by JmillsExpensify** [comment]
$500 payment approved for @sobitneupane based on BZ summary.
